src/algebra/category/Algebra/limits.lean

```diff
@@ -70,39 +87,103 @@ namespace has_limits
 Construction of a limit cone in `Algebra R`.
 (Internal use only; use the limits API.)
 -/
-def limit (F : J ⥤ Algebra R) : cone F :=
+def limit_cone (F : J ⥤ Algebra R) : cone F :=
 { X := Algebra.of R (limit (F ⋙ forget _)),
   π :=
   { app := limit_π_alg_hom F,
     naturality' := λ j j' f,
       alg_hom.coe_fn_inj ((limit.cone (F ⋙ forget _)).π.naturality f) } }
 
+@[simps]
+def forget_map_cone_limit_cone_iso (F : J ⥤ Algebra R) :
+  (forget (Algebra R)).map_cone (limit_cone F) ≅ limit.cone (F ⋙ forget (Algebra R)) :=
+{ hom := { hom := 𝟙 _, },
+  inv := { hom := 𝟙 _, } }
+
+def is_limit_forget_map_cone_limit_cone (F : J ⥤ Algebra R) :
+  is_limit ((forget (Algebra R)).map_cone (limit_cone F)) :=
+is_limit.of_iso_limit (limit.is_limit _) (forget_map_cone_limit_cone_iso F).symm
+
 /--
 Witness that the limit cone in `Algebra R` is a limit cone.
 (Internal use only; use the limits API.)
 -/
-def limit_is_limit (F : J ⥤ Algebra R) : is_limit (limit F) :=
+def limit_cone_is_limit (F : J ⥤ Algebra R) : is_limit (limit_cone F) :=
 begin
   refine is_limit.of_faithful
-    (forget (Algebra R)) (limit.is_limit _)
+    (forget (Algebra R)) (is_limit_forget_map_cone_limit_cone F)
     (λ s, { .. }) (λ s, rfl),
-  { simp only [forget_map_eq_coe, alg_hom.map_one, functor.map_cone_π], refl, },
-  { intros x y, simp only [forget_map_eq_coe, alg_hom.map_mul, functor.map_cone_π], refl, },
-  { simp only [forget_map_eq_coe, alg_hom.map_zero, functor.map_cone_π], refl, },
-  { intros x y, simp only [forget_map_eq_coe, alg_hom.map_add, functor.map_cone_π], refl, },
-  { intros r, ext j, dsimp, erw (s.π.app j).commutes r, refl, },
+  { intros, ext,
+    simp only [forget_map_eq_coe, forget_map_cone_limit_cone_iso_inv_hom, iso.symm_hom,
```

Comment:
This proof got a lot longer, but it's mainly huge `simp only` proofs.
Do I understand correctly that the general experience of proving things doesn't change much, `squeeze_simp` is taking care of the "heavy lifting"?

Reply:
This is mostly but not entirely true. The new and slightly unpleasant part of these proofs, which is not just being generated by automation, is the two steps `rw ←limit_π_alg_hom_apply, simp [limit_π_alg_hom],`. It works, but doens't feel very natural.